packages/detectors/src/detectors/server/request-hostname.ts

```diff
@@ -0,0 +1,18 @@
+import type { Locale } from '../../../../runtime/src/core'
+import type { LocaleDetector } from '../../detect'
+
+type Request = {
+	hostname: string
+}
+
+export const initRequestHostnameDetector =
+	(req: Request): LocaleDetector =>
+	(): Locale[] => {
+		const match = req?.hostname?.match(/^(?:([a-z]{2}(?:-[A-Z]{2})?)\.)/)
```

Comment:
Could you please store the regular expression in a variable at the root level of this file?
Recreating a regular expression each time a function gets executed could lead to performance issues.

Also: some languages use 3 digits. Can you please update the regex to account for 2 and 3 digits?
And please add some more tests to check multiple locale combinations

Reply:
I've extended the regular expression to match language tags as per RFC 5646, excluding extensions. The test suite for this addition uses the examples provided in Appendix A of the same.